packages/upgrade/scripts/upgrade.ts

```diff
@@ -0,0 +1,509 @@
+      "Upgrade",
+      rpc,
+      privateKey,
+      resume,
+      true
+    );
+
+    this.executeCommand(domainName, command, "upgrade-output");
+  }
+
+  setUpgradeEnv(domainName: string): void {
+    const config = this.nomadConfig;
+    const networks = config.networks;
+    const rpcs = config.rpcs;
+    const timelock =
+      config.protocol.networks[domainName].configuration.governance
+        .recoveryTimelock;
+
+    // Beacons for Core contracts
+    const homeBeacon = config.core[domainName].home.beacon;
+    const governanceRouterBeacon =
+      config.core[domainName].governanceRouter.beacon;
+
+    // Beacons for Bridge contracts
+    const bridgeRouterBeacon = config.bridge[domainName].bridgeRouter.beacon;
+    const tokenRegistryBeacon = config.bridge[domainName].tokenRegistry.beacon;
+    const bridgeTokenBeacon = config.bridge[domainName].bridgeToken.beacon;
+    // Get first replica beacon.
+    // All replicas in every domain, share the same beacon, as they share the same implementation
+    // but have different proxies, because they have different storage.
+    const replicaBeacon =
+      config.core[domainName].replicas[
+        Object.keys(config.core[domainName].replicas)[0]
+      ].beacon;
+
+    // UpgradeBeaconController and Governance Router
+    const upgradeBeaconController =
+      config.core[domainName].upgradeBeaconController;
+
+    // Set env variables to be picked up by forge script
+    // Set beacon addresses
+    process.env["NOMAD_HOME_BEACON"] = homeBeacon;
+    process.env["NOMAD_GOVERNANCE_ROUTER_BEACON"] = governanceRouterBeacon;
+    process.env["NOMAD_BRIDGE_ROUTER_BEACON"] = bridgeRouterBeacon;
+    process.env["NOMAD_TOKEN_REGISTRY_BEACON"] = tokenRegistryBeacon;
+    process.env["NOMAD_BRIDGE_ROUTER_BEACON"] = bridgeTokenBeacon;
+    process.env["NOMAD_REPLICA_BEACON"] = replicaBeacon;
+    // set env variable for timelock
+    process.env["NOMAD_RECOVERY_TIMELOCK"] = timelock.toString();
+    process.env["NOMAD_BEACON_CONTROLLER"] = upgradeBeaconController;
+  }
+
+  upgradeForkTest(domainName: string): void {
+    console.log("Starting Fork test...");
+    console.log(
+      "Results and output will be printed after the fork test is complete"
+    );
+    // Get first replica beacon.
+    // All replicas in every domain, share the same beacon, as they share the same implementation
+    // but have different proxies, because they have different storage.
+    const config = this.nomadConfig;
+    const rpc = process.env.RPC_URL || config.rpcs[domainName][0];
+    const replicaProxy =
+      config.core[domainName].replicas[
+        Object.keys(config.core[domainName].replicas)[0]
+      ].proxy;
+
+    const governanceRouterProxy =
+      config.core[domainName].governanceRouter.proxy;
+
+    // Set env variables required by Fork test and Upgrade
+    process.env["NOMAD_REPLICA_PROXY"] = replicaProxy;
+    process.env["NOMAD_GOV_ROUTER_PROXY"] = governanceRouterProxy;
```

Comment:
We don't need those because the `forkTest` only tests a particular part of the upgrade process, that is the change we made to message status. 

We don't test the upgraded contracts because that should be covered by the normal unit tests, IMHO.

ForkTest should only test things that we expect to break in relation to before/after.  We should expand the forkTest tests for sure